apps/server/src/modules/board/service/board-do.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { AnyBoardDo, EntityId } from '@shared/domain';
+import { BoardDoRepo } from '../repo';
+
+@Injectable()
+export class BoardDoService {
+	constructor(private readonly boardDoRepo: BoardDoRepo) {}
+
+	async findById(id: EntityId): Promise<AnyBoardDo> {
+		const domainObject = await this.boardDoRepo.findById(id);
+		return domainObject;
+	}
+
+	async findParentOfId(childId: EntityId): Promise<AnyBoardDo | undefined> {
+		const parent = await this.boardDoRepo.findParentOfId(childId);
+		return parent;
+	}
+
+	async deleteChild<T extends AnyBoardDo>(parent: T, childId: EntityId): Promise<T> {
```

Comment:
I think these kinds of operations should be done in the domain objects (aggregates).